### PR TITLE
Task 006/upgrade c make lists (issue #6)

### PR DESCRIPTION
**Pull request by GRPilot**
I also add some comments in CMakeLists and *.cmake files as asked [here](https://github.com/GRPilot/lazycoder/projects/1#card-66461140)

Now you can forget about changing something in CMakeLists because CMake collects all your new files and directories themselves.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,65 +1,54 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.5) # The smallest abailable cmake version for use
```
Comment (seigtm) [inline CMakeLists.txt:1]: available

**Review by seigtm — CHANGES_REQUESTED** [review]
Check CMakeLists. I left commentary there

File: CMakeLists.txt (hunk 1 above)
Comment (seigtm) [reply CMakeLists.txt:1]: "to use"

Comment (GRPilot) [reply CMakeLists.txt]: fixed in commit 08f5be3

**Review by seigtm — APPROVED** [review]

**Pull request closed without merging**